plaid/react-native-plaid-link-sdk
# [Feat] Expand iOS Presentation Methods

## Summary


Adds the ability for Link to be presented using [UIModalPresentationOverFullScreen](https://developer.apple.com/documentation/uikit/uimodalpresentationstyle/uimodalpresentationoverfullscreen?language=objc).

## Motivation


Customer request.

## :pencil: Checklist
- [x] I have optimized code readability (class/variable names, straight forward logic paths, short clarifying docs,...).

## :green_heart: Testing


You will need to [test the local changes using the example app](https://github.com/plaid/react-native-plaid-link-sdk/blob/master/CONTRIBUTING.md#testing-local-changes). 

1. run `yalc publish` at the root of the project
2. `cd example`
3. run `yalc add react-native-plaid-link-sdk`
4. `npm install`
5. `cd ios`
6. `bundle install`
7. `bundle exec pod install`
8. `npx react-native start`
9. `npx react-native run-ios`
10. See the example app present Link as a modal
11. Change line 37 in `PlaidEventContainser.tsx` from `iOSPresentationStyle: LinkIOSPresentationStyle.MODAL,` to `iOSPresentationStyle: LinkIOSPresentationStyle.FULL_SCREEN,`.
12. Launch link again and see it presented full screen.

**Example**:

https://github.com/plaid/react-native-plaid-link-sdk/assets/122908338/06e09269-ee2b-4190-8024-5d4b9c05511d


## Documentation

Select one:
 
- [x] I have added relevant documentation for my changes.
- [ ] This PR does not result in any developer-facing changes.
